File: src/types/session.ts
export type ToolType = 'claude' | 'codex';

export type SessionStatus = 'running' | 'idle' | 'active' | 'error' | 'done';

export interface SessionInfo {
  id: string;
  tool: ToolType;
  cwd: string;
  label: string;
  status: SessionStatus;
  created_at: number;
  position: { x: number; y: number };
  isGhost?: boolean;
  taskTitle?: string;
  previewUrl?: string;
  /** tmux session name (e.g. "vcc-<uuid>") — present when session is tmux-backed */
  tmuxName?: string;
  /** True when this ghost tile has a live tmux session that can be reattached */
  isLiveGhost?: boolean;
}

/** Info about a discovered tmux session (returned from backend) */
export interface TmuxSessionInfo {
  tmux_name: string;
  session_id: string;
}

export interface CreateSessionRequest {
  tool: ToolType;
  cwd: string;
  initial_prompt?: string;
  taskTitle?: string;
  previewUrl?: string;
}

// --- Output Intelligence Types ---

export interface QuestionInfo {
  text: string;
  type: 'yn' | 'permission' | 'open';
  detectedAt: number;
}

export interface OutcomeInfo {
  summary: string;
  filesChanged?: number;
  duration: number;
}

export interface SessionIntel {
  lastActivity: string;
  detectedUrl: string | null;
  pendingQuestion: QuestionInfo | null;
  outcome: OutcomeInfo | null;
}

File: src-tauri/src/session.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ToolType {
    #[serde(rename = "claude")]
    Claude,
    #[serde(rename = "codex")]
    Codex,
}

impl ToolType {
    pub fn command(&self) -> &str {
        match self {
            ToolType::Claude => "claude",
            ToolType::Codex => "codex",
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum SessionStatus {
    #[serde(rename = "running")]
    Running,
    #[serde(rename = "idle")]
    Idle,
    #[serde(rename = "active")]
    Active,
    #[serde(rename = "error")]
    Error,
    #[serde(rename = "done")]
    Done,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionInfo {
    pub id: String,
    pub tool: ToolType,
    pub cwd: String,
    pub label: String,
    pub status: SessionStatus,
    pub created_at: u64,
}
